### PR TITLE
Apache Pulsar Certification Tests

### PR DESCRIPTION
**Pull request by addjuarez**
Signed-off-by: addjuarez <6789375+addjuarez@users.noreply.github.com>

# Description

Addition of Apache Pulsar Certification Tests

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

#2013 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: tests/certification/pubsub/pulsar/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   - Verify the creation of topic
+   - Send messages to the topic created
+   - Verify that subscriber received all the messages
+- Verify reconnection after the network interruptions
+   - Run dapr application with 1 publisher and 1 subscriber
+   - Publisher publishes to 1 topic
+   - Subscriber is subscribed to 1 topic
+   - Simulate network interruptions and verify that the component retires on error
+   - Verify that all expected messages were received
+- Verify data with an optional parameter delayqueue set
```
Comment (berndverst) [inline tests/certification/pubsub/pulsar/README.md:42]: you are not actually using this parameter anywhere so your current test should not pass.. if it does your test isn't correct

**Review by berndverst — CHANGES_REQUESTED** [review]
Lots of spelling mistakes - take a closer look :)

And fix the `delayqueue` test per my comment. After that I'll take another look.

File: tests/certification/pubsub/pulsar/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+version: '2'
+services:
+  standalone:
+    image: apachepulsar/pulsar:2.8.2
+    ports:
+      - "6650:6650"
+      - "8080:8080"
+    environment:
+      - BOOKIE_MEM=" -Xms768m -Xms768m -XX:MaxDirectMemorySize=1g"
```
Comment (DeepanshuA) [inline tests/certification/pubsub/pulsar/docker-compose.yml:9]:
```suggestion
      - BOOKIE_MEM=" -Xms768m -Xms768m -XX:MaxDirectMemorySize=1g"
```
- BOOKIE_MEM=" -Xms768m -Xmx768m -XX:MaxDirectMemorySize=1g"

File: tests/certification/pubsub/pulsar/pulsar_test.go [hunk 3]
```diff
@@ -0,0 +1,1180 @@
...
+func TestPulsarNetworkInterruption(t *testing.T) {
+	consumerGroup1 := watcher.NewUnordered()
+
+	// Set the partition key on all messages so they are written to the same partition. This allows for checking of ordered messages.
+	metadata := map[string]string{
+		messageKey: partition0,
+	}
+
+	// subscriber of the given topic
+	subscriberApplication := func(appID string, topicName string, messagesWatcher *watcher.Watcher) app.SetupFn {
```
Comment (DeepanshuA) [inline tests/certification/pubsub/pulsar/pulsar_test.go:786]: `subscriberApplication`, `publishMessages` and `assertMessages` seems to be almost same in all the places - will it be possible to take them out in a separate function?

**Review by berndverst — CHANGES_REQUESTED** [review]
The `delayqueue` property is still not being used.

**Comment by addjuarez** [comment]
> The `delayqueue` property is still not being used.

My mistake, delayqueue isnt a property. The properties should be deliverAt and deliverAfter. These are also metadata query parameters and not component properties.

File: .github/workflows/certification.yml [hunk 4]
```diff
@@ -50,6 +50,8 @@ jobs:
         PR_COMPONENTS=$(yq -I0 --tojson eval - << EOF
         - pubsub.kafka
         - pubsub.rabbitmq
+        - pubsub.mqtt
```
Comment (berndverst) [inline .github/workflows/certification.yml:53]:
```suggestion
```

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
@addjuarez  Open docs PR again 1.10 branch to make this component stable.

**Pull request merged**